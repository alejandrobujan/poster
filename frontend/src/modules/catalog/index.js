import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as SearchBar} from "./components/SearchBar";
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


