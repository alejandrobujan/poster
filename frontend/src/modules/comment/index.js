import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Comments} from './components/Comments';
export {default as Comment} from './components/Comment';

// eslint-disable-next-line
export default  {actions, reducer, selectors};