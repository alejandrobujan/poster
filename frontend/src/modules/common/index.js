export {default as Pager} from "./components/Pager";
export {default as Errors} from "./components/Errors";
export {default as BackLink} from "./components/BackLink";
export {default as UserCard} from "./components/UserCard";
export {default as PostLink} from "./components/PostLink";
export {default as ImagesCard} from "./components/ImagesCard";