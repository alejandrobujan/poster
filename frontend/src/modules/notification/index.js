import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Notifications} from "./components/Notifications";
export {default as Notification} from "./components/Notification";

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


