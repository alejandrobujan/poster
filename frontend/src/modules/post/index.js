import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as CreatePost} from './components/CreatePost';
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';
export {default as PostDetails} from './components/PostDetails';


// eslint-disable-next-line
export default  {actions, reducer, selectors};