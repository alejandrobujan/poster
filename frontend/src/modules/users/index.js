import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Login} from './components/Login';
export {default as SignUp} from './components/SignUp';
export {default as Logout} from './components/Logout';
export {default as UpdateProfile} from './components/UpdateProfile';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};