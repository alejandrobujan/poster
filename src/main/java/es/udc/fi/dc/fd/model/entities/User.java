package es.udc.fi.dc.fd.model.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

/**
 * The Class User.
 */
@Entity
@Table(name = "Users")
public class User {

	/**
	 * The Enum RoleType.
	 */
	public enum RoleType {
		/** The user. */
		USER
	}

	/** The id. */
	private Long id;

	/** The user name. */
	private String userName;

	/** The password. */
	private String password;

	/** The first name. */
	private String firstName;

	/** The last name. */
	private String lastName;

	/** The email. */
	private String email;

	/** The avatar. */
	private byte[] avatar;

	/** The role. */
	private RoleType role;

	/**
	 * Instantiates a new user.
	 */
	public User() {
	}

	/**
	 * Instantiates a new user.
	 *
	 * @param userName  the user name
	 * @param password  the password
	 * @param firstName the first name
	 * @param lastName  the last name
	 * @param email     the email
	 * @param avatar    the avatar
	 */
	public User(String userName, String password, String firstName, String lastName, String email, byte[] avatar) {

		this.userName = userName;
		this.password = password;
		this.firstName = firstName;
		this.lastName = lastName;
		this.email = email;
		this.avatar = avatar;

	}

	/**
	 * Gets the id.
	 *
	 * @return the id
	 */
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	public Long getId() {
		return id;
	}

	/**
	 * Sets the id.
	 *
	 * @param id the new id
	 */
	public void setId(Long id) {
		this.id = id;
	}

	/**
	 * Gets the user name.
	 *
	 * @return the user name
	 */
	public String getUserName() {
		return userName;
	}

	/**
	 * Sets the user name.
	 *
	 * @param userName the new user name
	 */
	public void setUserName(String userName) {
		this.userName = userName;
	}

	/**
	 * Gets the password.
	 *
	 * @return the password
	 */
	public String getPassword() {
		return password;
	}

	/**
	 * Sets the password.
	 *
	 * @param password the new password
	 */
	public void setPassword(String password) {
		this.password = password;
	}

	/**
	 * Gets the first name.
	 *
	 * @return the first name
	 */
	public String getFirstName() {
		return firstName;
	}

	/**
	 * Sets the first name.
	 *
	 * @param firstName the new first name
	 */
	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	/**
	 * Gets the last name.
	 *
	 * @return the last name
	 */
	public String getLastName() {
		return lastName;
	}

	/**
	 * Sets the last name.
	 *
	 * @param lastName the new last name
	 */
	public void setLastName(String lastName) {
		this.lastName = lastName;
	}

	/**
	 * Gets the email.
	 *
	 * @return the email
	 */
	public String getEmail() {
		return email;
	}

	/**
	 * Sets the email.
	 *
	 * @param email the new email
	 */
	public void setEmail(String email) {
		this.email = email;
	}

	/**
	 * Gets the avatar image.
	 * 
	 * @return the avatar
	 */
	public byte[] getAvatar() {
		return avatar;
	}

	/**
	 * Sets the avatar image
	 * 
	 * @param avatar the avatar to set
	 */
	public void setAvatar(byte[] avatar) {
		this.avatar = avatar;
	}

	/**
	 * Gets the user role
	 * 
	 * @return the role
	 */
	public RoleType getRole() {
		return role;
	}

	/**
	 * Sets the user role
	 * 
	 * @param role the role to set
	 */
	public void setRole(RoleType role) {
		this.role = role;
	}

	@Override
	public String toString() {
		return "User [id=" + id + ", userName=" + userName + ", password=" + password + ", firstName=" + firstName
				+ ", lastName=" + lastName + ", email=" + email + ", role=" + role + "]";
	}

}
