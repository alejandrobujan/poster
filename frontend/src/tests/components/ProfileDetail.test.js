import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import {Provider} from 'react-redux';

import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import { userMock } from "../mocks/User.mock";

import renderer from "react-test-renderer";

import { ProfileDetail } from "../../modules/users";

describe("ProfileDetails", () => {
	const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);
    let store;    
    store = mockStore(userMock);
      

    it("renders correctly", () => {
		const tree = renderer
        .create(
				<Provider store={store}>
					<MemoryRouter>
						<ProfileDetail/>
					</MemoryRouter>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	
	it("basic structure", () => {
        render(
			<Provider store={store}>
				<MemoryRouter>
                    <ProfileDetail />
				</MemoryRouter>
			</Provider>
		);
		
		const avatar = screen.getByTestId('Avatar');
		const userName = screen.getByTestId('UserName');
		const firstName = screen.getByTestId('FirstName');
		const lastName = screen.getByTestId('LastName');
		const email = screen.getByTestId('Email');
		const editButton = screen.getByTestId('EditButton');
		
	});
	
	it("click edit profile button", async () => {
        render(
			<Provider store={store}>
				<MemoryRouter>
                    <ProfileDetail />
				</MemoryRouter>
			</Provider>
		);
		
		const editProfileButton = screen.getByTestId('EditButton');

        fireEvent.click(editProfileButton);		
	});
	
});