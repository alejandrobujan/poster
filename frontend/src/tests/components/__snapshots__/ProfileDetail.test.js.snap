// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileDetails renders correctly 1`] = `
<div>
  <div
    className="card text-center"
  >
    <div
      className="card-body row"
    >
      <div
        className="col"
      >
        <img
          alt="Avatar"
          data-testid="Avatar"
          height="200px"
          src="/poster/assets/profile.png"
          width="200px"
        />
        Â 
        <h5
          className="card-title"
          data-testid="UserName"
        />
      </div>
      <div
        className="col text-left"
      >
        <ul
          className="list-group list-group-flush"
        >
          <li
            className="list-group-item"
            data-testid="FirstName"
          >
            First Name: 
            <b />
          </li>
          <li
            className="list-group-item"
            data-testid="LastName"
          >
            Last Name: 
            <b />
          </li>
          <li
            className="list-group-item"
            data-testid="Email"
          >
            Email: 
            <b />
          </li>
        </ul>
      </div>
    </div>
    <div
      className="text-center mb-4"
    >
      <button
        className="btn btn-primary"
        data-testid="EditButton"
        onClick={[Function]}
        type="button"
      >
        Edit Profile
      </button>
    </div>
  </div>
</div>
`;
